src/ostree/ot-builtin-prune.c

```diff
@@ -255,20 +255,19 @@ ostree_builtin_prune (int argc, char **argv, GCancellable *cancellable, GError *
           for (char **iter = opt_only_branches; iter && *iter; iter++)
             {
               const char *ref = *iter;
-              g_autofree char *commit = NULL;
               /* Ensure the specified branch exists */
-              if (!ostree_repo_resolve_rev (repo, ref, FALSE, &commit, error))
+              if (!ostree_repo_resolve_rev (repo, ref, FALSE, NULL, error))
                 return FALSE;
-              g_hash_table_add (only_branches_set, *iter);
+              if (g_hash_table_contains (retain_branch_depth, ref))
+                return glnx_throw (error, "--retain-branch-depth conflicts with --only-branch");
```

Comment:
We want this bit down there no? I.e. we do want to be able to specify `--retain-branch-depth=my-ref=0 --only-branch my-ref`, right?

I was targeting the `g_hash_table_insert` below instead:

```c
GLNX_HASH_TABLE_FOREACH (all_refs, const char *, ref)
  {
    if (!g_hash_table_contains (only_branches_set, ref))
      {
        if (!g_hash_table_insert (retain_branch_depth, g_strdup (ref), GINT_TO_POINTER ((int)-1)))
          return glnx_throw (error, "--retain-branch-depth conflicts with --only-branch");
      }
  }
```

Reply:
Hum...so a high level design I had here was that with `--only-branch` one just uses the global `--depth`.  In other words: `--retain-branch-depth=my-ref=0 --only-branch my-ref` == `--refs-only --only-branch my-ref` right?

I'm not quite sure when one would want to combine `--only-branch` with different depths.

Reply:
Don't we want to support e.g. `--only-branch foo --only-branch bar` ? So then you have to resort to `--retain-branch-depth` if you want them pruned at different depths, no? Well, they can just use separate invocations I suppose (or use the API directly). Though it seems easy enough to support. Otherwise, we should just error out if any `--retain-branch-depth` is passed at all when `--only-branch` is given I'd say. And also, remove the related comment block:

```diff
+      /* Process --only-branch. Note this combines with --retain-branch-depth; one
+       * could do e.g.:
+       *  * --only-branch exampleos/x86_64/foo
+       *  * --retain-branch-depth exampleos/x86_64/foo=0
+       * to prune exampleos/x86_64/foo to just the latest commit.
+       */
```

Reply:
Yeah, I think your original comment here was right - I don't see a reason not to support combining the two, even with different branches.  Thanks for sticking this out...another fixup ⬇️